### ISSUE TITLE
md-spinner freezes during javascript execution

### ISSUE THREAD

**Issue by fallXone** [angular/material2#6094]
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
md-spinner is unaffected by javascript execution 

#### What is the current behavior?
md-spinner locks up during any sort of javascript execution. Sometimes it slows down during execution as well, and the spinner will look different (like a small circle going around)

#### What are the steps to reproduce?
Execute javascript code while the spinner is visible (see plnkr)
https://plnkr.co/edit/dwIOBUploDOpy8tvMBvd?p=preview

#### What is the use-case or motivation for changing an existing behavior?
Usability - the spinner should spin while something is happening. A frozen spinner doesn't indicate that something is loading.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular: 4.23
Material: 2.0.0-beta.8
OS: Windows 8.1

**Comment by andrewseguin** [comment]
This is something we are looking at to fix. The spinner should rely on CSS rather than javascript for its animation so it isnt blocked on the main thread.

**Comment by fallXone** [comment]
Is there any update on the progress for this? It's somewhat urgent on our team or we will have to resort to another spinner, as not only does it sometimes freeze, but sometimes the animation gets truncated into just a dot and looks very awkward.

**Comment by siebmanb** [comment]
workaround: I temporarily replaced it with a gif spinner built online, it was super easy and spinner is almost identical. I hid svg part of md-spinner with css and added my gif as a background.

**Comment by fallXone** [comment]
I've tried a .gif spinner and unfortunately had the same result for some reason

**Comment by russelltrafford** [comment]
I hope this gets reclassified. For us it's not minor, as the loading icons jerk all over the place during async content loads, which defeats the whole idea of smooth experience in a single page app.

Might have to resort to a basic CSS only spinner in the meantime, as this was raised back in July, and its now Nov, so looks like it will be a long time coming.

Through my testing, can confirm that the CSS only spinners I tried don't freeze during JS.

**Comment by jelbourn** [comment]
The spinner is now totally css-based. Have you tried it with the latest release?

**Comment by samqhr** [comment]
If the spinner is CSS based, then why is the ticket still open? Does this resolve the issue, and which version resolved the issue?

**Comment by samqhr** [comment]
Update: Looks like fixed version was 2.0.0-beta.12 marble-mustache according to the changelog  https://github.com/angular/material2/blob/master/CHANGELOG.md

**Comment by jelbourn** [comment]
Closing as fixed. Feel free to open a new issue if something is still not behaving correctly.

**Comment by russelltrafford** [comment]
Ok thanks for the update @jelbourn and @samqhr for checking the changelog updates
